packages/ipfs-core/src/components/files/utils/hamt-utils.js

```diff
@@ -62,13 +76,14 @@ const addLinksToHamtBucket = async (links, bucket, rootBucket) => {
         const pos = parseInt(link.Name, 16)
 
         bucket._putObjectAt(pos, new Bucket({
-          hashFn: DirSharded.hashFn
+          hash: rootBucket._options.hash,
+          bits: rootBucket._options.bits
         }, bucket, pos))
 
         return Promise.resolve()
       }
 
-      return (rootBucket || bucket).put(link.Name.substring(2), {
+      return rootBucket.put(link.Name.substring(2), {
```

Comment:
I'm not seeing the guarantee in the changes here that there _will_ be a `rootBucket` at this point. It's added in `recreateHamtLevel` if there's not a `parentBucket`, but are we sure that there's always a `rootBucket` if there's a `parentBucket`? A quick `git grep` suggests that `recreateHamtLevel` either has only the first arg or all of the args, but I don't know whether those args always have a non-null value.